src/OpenTracing/Util/GlobalTracer.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using OpenTracing.Propagation;
+
+namespace OpenTracing.Util
+{
+    /// <summary>
+    /// Global tracer that forwards all methods to another tracer that can be
+    /// configured by calling <see cref="Register(ITracer)"/>.
+    /// <para/>
+    /// The <see cref="Register(ITracer)"/> method should only be called once
+    /// during the application initialization phase.<br/>
+    /// If the <see cref="Register(ITracer)"/> method is never called,
+    /// the default <see cref="NoopTracer"/> is used.
+    /// <para/>
+    /// Where possible, use some form of dependency injection (of which there are
+    /// many) to access the <see cref="ITracer"/> instance. For vanilla application code, this is
+    /// often reasonable and cleaner for all of the usual DI reasons.
+    /// <para/>
+    /// That said, instrumentation for packages that are themselves statically
+    /// configured (e.g., JDBC drivers) may be unable to make use of said DI
+    /// mechanisms for <see cref="ITracer"/> access, and as such they should fall back on
+    /// <see cref="GlobalTracer"/>. By and large, OpenTracing instrumentation should
+    /// always allow the programmer to specify a <see cref="ITracer"/> instance to use for
+    /// instrumentation, though the <see cref="GlobalTracer"/> is a reasonable fallback or
+    /// default value.
+    /// </summary>
+    public sealed class GlobalTracer : ITracer
+    {
+        private static readonly object s_lock = new object();
+
+        /// <summary>
+        /// Singleton instance.
+        /// <para/>
+        /// Since we cannot prevent people using <see cref="Instance"/> as a constant,
+        /// this guarantees that references obtained before, during or after initialization
+        /// all behave as if obtained <em>after</em> initialization once properly initialized.<br/>
+        /// As a minor additional benefit it makes it harder to circumvent the <see cref="ITracer"/> API.
+        /// </summary>
+        private static readonly GlobalTracer s_instance = new GlobalTracer();
+
+        /// <summary>
+        /// Returns the constant <see cref="GlobalTracer"/>.
+        /// <para/>
+        /// All methods are forwarded to the currently configured tracer.<br/>
+        /// Until a tracer is explicitly configured via <see cref="Register(ITracer)"/>,
+        /// the <see cref="NoopTracer"/> is used.
+        /// </summary>
+        /// <returns>The global tracer constant.</returns>
+        /// <seealso cref="Register(ITracer)"/>
+        public static ITracer Instance => s_instance;
```

Comment:
Minor concern, but could add up.
Since this is itself exposed as an `ITracer` rather than it's real type, it will result in all calls to it like `BuildSpan` becoming `CallVirt`'s. Then, since it wraps another, it has another `CallVirt`.

I really like information hiding, but in this case exposing it's real type may be best since it's likely all tracing will come through this class.

Reply:
Hmmm... no idea what the perf impact is here. I'll do a little bit of research. Returning `GlobalTracer` probably means that IntelliSense no longer shows documentation since `<inheritdoc/>` is not really supported in IDEs AFAIK. We would have to copy all the documentation.

Reply:
It looks like something that could be done afterwards as an optimization (an issue around this could be created so it doesn't get forgotten, and eventually discussed/handled).